libraries/AP_Common/Location.h

```diff
@@ -15,6 +15,7 @@ class Location
     uint8_t terrain_alt  : 1;           // this altitude is above terrain
     uint8_t origin_alt   : 1;           // this altitude is above ekf origin
     uint8_t loiter_xtrack : 1;          // 0 to crosstrack from center of waypoint, 1 to crosstrack from tangent exit location
+    uint8_t local_frame :1;             // 1 if lat/lng is N/E postion vector in centimeters relative to ekf origin
```

Comment:
I don't think we should need to add the local_frame field to the Location object.   Instead we should probably try and figure out from the mission item whether it is using lat/lon or local coordinates and if local it should be put into a Vector3f and send to the AC_WPNav library.

Reply:
Thank you @rmackay9 
In function write_cmd_to_storage()
https://github.com/ArduPilot/ardupilot/blob/e4d510718364738c6bf749a1052f14f2c8075037/libraries/AP_Mission/AP_Mission.cpp#L732
It passed Mission_Command, but there is not possible to tell whether it is using lat/lon or local coordinates without add a new field

Reply:
@chobitsfan, thanks for all this.

@peterbarker mentioned something about creating a new method to read and/or write commands that uses a Vector3f instead of a Location but I must admit that I didn't really understand what he meant because I'm not very familiar with this part of the code.

I think though that the need to add the "local_frame" to the Location class exposes the core issue which is that the code has a built in assumption that command's positions must use lat,lon.  We are attempting to break this assumption which means we need some lower level changes in how mission commands are stored.

Reply:
Hi @rmackay9 Thank you

> @peterbarker mentioned something about creating a new method to read and/or write commands that uses a Vector3f instead of a Location

Yes. I agree with you. But I think it seems not possible to do it in current mission storage limit. Because stored_in_location() is based on cmd id. 
https://github.com/ArduPilot/ardupilot/blob/94c4b2d7698d3780135241c1cb5e561e5174a332/libraries/AP_Mission/AP_Mission.cpp#L690
We can not tell MAV_CMD_NAV_WAYPOINT is stored in vector3f or Location in current mission storage. I think I will close this PR and wait for new mission storage.